processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -184,6 +203,7 @@ protected void scanAndAggregate(
 
     final int aggSize = theAggregators.length;
     final int aggExtra = aggSize % AGG_UNROLL_COUNT;
+    final AtomicInteger currentPosition = new AtomicInteger(0);
```

Comment:
how about not incrementing it within the method, but instead make sure each call of aggregateDimValue is followed by  `currentPosition++`? Can we benchmark both and see what the impact is?